music_this_week_app/views.py

```diff
@@ -79,23 +80,35 @@ def callback(request):
 
 def setup(request):
     """Displays search args and a search button, which links to /search"""
-
+    #get user ip address to find user location 
+    x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR')
+    if x_forwarded_for:
+        ip = x_forwarded_for.split(',')[0]
+    else:
+        ip = request.META.get('REMOTE_ADDR')
+    #handle invalid IP or ip == 127.0.0.1 because running locally 
+    try:
+        g = GeoIP2()
+        location = g.city(ip)
```

Comment:
Very minor comment, but it seems like the fact that `location` is a dict necessitates a little bit of acrobatics further down, and just made it a little bit harder to understand what is what.

What if this line were `location = g.city(ip).get('city')`? Then `location` would be a string that would fit more cleanly later on. Or are there other items in the `location` dict that you use or may use in the future?